### PR TITLE
Add CI/CD for Docker Build and Sonar cloud

### PR DESCRIPTION
**Pull request by SaikrishnaBairamoni**
This PR is to add the GitHub actions workflows for both docker hub build and to run sonar scan analysis for publishing static reports on sonar cloud dashboard.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "temurin"
+      - name: Run Sonar
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          ls -la && pwd
+          mvn -e -X clean org.jacoco:jacoco-maven-plugin:prepare-agent package sonar:sonar -Dsonar.projectKey=usdot-jpo-ode_jpo-sdw-depositor -Dsonar.projectName=jpo-sdw-depositor -Dsonar.organization=usdot-jpo-ode-1 -Dsonar.host.url=https://sonarcloud.io -Dsonar.branch.name=$GITHUB_REF_NAM
```
Comment (codygarver) [inline .github/workflows/ci.yml:30]: You should be able to get rid of many of these flags since they are duplicated in `pom.xml`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          submodules: recursive
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "temurin"
+      - name: Run Sonar
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
```
Comment (codygarver) [inline .github/workflows/ci.yml:28]: Unit tests are not being run so no chance for coverage report to be generated. The README says `mvn test` is supported, need to incorporate that into CI and get it to output a coverage report for sonar to import.

**Review by codygarver — CHANGES_REQUESTED** [review]
In addition to diff comments, be sure to delete old CI `.travis.yml`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          path: home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor/target           
+      - name: Setup SonarScanner
+        uses: warchant/setup-sonar-scanner@v4
+        with:
+          version: 4.8.0.2856
+      - name: Generate sonar properties file
+        run: |
+          cat <<EOF > /tmp/sonar-scanner.properties
+          sonar.host.url=https://sonarcloud.io
+          sonar.coverage.jacoco.xmlReportPaths=/home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor/home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor/target/site/jacoco/jacoco.xml
```
Comment (codygarver) [inline .github/workflows/ci.yml:56]: This path looks like it was double copy pasted

File: pom.xml [hunk 4]
```diff
@@ -17,6 +17,14 @@
...
    <properties>
       <jmockit.version>1.49</jmockit.version>
+      <sonar.organization>usdot-jpo-ode-1</sonar.organization>
+      <sonar.host.url>https://sonarcloud.io</sonar.host.url>
+      <!-- JaCoCo Properties -->
+      <jacoco.version>0.8.8</jacoco.version>
+      <sonar.java.coveragePlugin>jacoco</sonar.java.coveragePlugin>
+      <sonar.dynamicAnalysis>reuseReports</sonar.dynamicAnalysis>
+      <sonar.coverage.jacoco.xmlReportPaths>${project.basedir}/target/site/jacoco/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
+      <sonar.language>java</sonar.language>
```
Comment (codygarver) [inline pom.xml:27]: Is this being used? It's duplicating stuff in the sonar properties file like organization

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Checkout ${{ github.event.repository.name }}
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0   
+      - run: chmod +x -R /home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor
+      - name: Download code coverage results
+        uses: actions/download-artifact@v3
+        with:
+          name: jpo-sdw-depositor
+          path: home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor/target
```
Comment (codygarver) [inline .github/workflows/ci.yml:47]: This path looks like a typo

**Review by codygarver — CHANGES_REQUESTED** [review]
The sonar step log indicates no coverage report is being scanned

`2023-06-08T20:21:25.2446522Z 20:21:25.244 WARN: No coverage report can be found with sonar.coverage.jacoco.xmlReportPaths='/home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor/home/runner/work/jpo-sdw-depositor/jpo-sdw-depositor/target/site/jacoco/jacoco.xml'. Using default locations: target/site/jacoco/jacoco.xml,target/site/jacoco-it/jacoco.xml,build/reports/jacoco/test/jacocoTestReport.xml`

I don't see any jacoco files in the artifact either, does this project use a different coverage type/file maybe?

File: pom.xml [hunk 6]
```diff
@@ -97,10 +107,31 @@
             <dependency>
               <groupId>org.apache.maven.surefire</groupId>
               <artifactId>surefire-junit47</artifactId>
-              <version>3.0.0-M5</version>
+              <version>3.0.0-M7</version>
             </dependency>
-          </dependencies>
+          </dependencies>
```
Comment (codygarver) [inline pom.xml:112]: Should probably revert these changes?

**Review by codygarver — CHANGES_REQUESTED** [review]
See diff comments

File: pom.xml [hunk 7]
```diff
@@ -84,10 +93,11 @@
          <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
-          <version>3.0.0-M5</version>
+          <version>3.0.0-M7</version>
           <configuration>
-            <argLine>-javaagent:${user.home}/.m2/repository/org/jmockit/jmockit/${jmockit.version}/jmockit-${jmockit.version}.jar</argLine>
             <!-- <testFailureIgnore>true</testFailureIgnore> -->
+                <enableProcessChecker>all</enableProcessChecker>
+                <shutdown>exit</shutdown>
```
[the remark this answers is not in the document]
Comment (SaikrishnaBairamoni) [reply pom.xml:100]: updated the version as the older version of 3.0.0-M5 is failing

File: pom.xml (hunk 6 above)
Comment (SaikrishnaBairamoni) [reply pom.xml:112]: updated the version as the older version of 3.0.0-M5 is failing

**Review by codygarver — APPROVED** [review]
